### ISSUE TITLE
UserID doesn't exist in example

### ISSUE THREAD

**Issue by victor-ono** [ClickHouse/clickhouse-docs#625]
[https://clickhouse.com/docs/en/guides/improving-query-performance/sparse-primary-indexes/sparse-primary-indexes-intro/](url)

```
SELECT URL, count(URL) as Count
FROM hits_NoPrimaryKey
WHERE UserID = 749927693
GROUP BY URL
ORDER BY Count DESC
LIMIT 10;
```

`0 rows in set. Elapsed: 0.021 sec. Processed 8.87 million rows, 35.47 MB (421.67 million rows/s., 1.69 GB/s.)`

**Comment by DanRoscigno** [comment]
@tom-clickhouse can we swap out the existing userid in the docs with `967801418`?

**Comment by DanRoscigno** [comment]
Thanks @victor-ono !  The dataset must have been updated.

**Comment by tom-clickhouse** [comment]
Can't reproduce it:

```
CREATE TABLE hits_NoPrimaryKey
(
    `UserID` UInt32,
    `URL` String,
    `EventTime` DateTime
)
ENGINE = MergeTree
PRIMARY KEY tuple();


INSERT INTO hits_NoPrimaryKey SELECT
   intHash32(UserID) AS UserID,
   URL,
   EventTime
FROM url('https://datasets.clickhouse.com/hits/tsv/hits_v1.tsv.xz', 'TSV', 'WatchID UInt64,  JavaEnable UInt8,  Title String,  GoodEvent Int16,  EventTime DateTime,  EventDate Date,  CounterID UInt32,  ClientIP UInt32,  ClientIP6 FixedString(16),  RegionID UInt32,  UserID UInt64,  CounterClass Int8,  OS UInt8,  UserAgent UInt8,  URL String,  Referer String,  URLDomain String,  RefererDomain String,  Refresh UInt8,  IsRobot UInt8,  RefererCategories Array(UInt16),  URLCategories Array(UInt16), URLRegions Array(UInt32),  RefererRegions Array(UInt32),  ResolutionWidth UInt16,  ResolutionHeight UInt16,  ResolutionDepth UInt8,  FlashMajor UInt8, FlashMinor UInt8,  FlashMinor2 String,  NetMajor UInt8,  NetMinor UInt8, UserAgentMajor UInt16,  UserAgentMinor FixedString(2),  CookieEnable UInt8, JavascriptEnable UInt8,  IsMobile UInt8,  MobilePhone UInt8,  MobilePhoneModel String,  Params String,  IPNetworkID UInt32,  TraficSourceID Int8, SearchEngineID UInt16,  SearchPhrase String,  AdvEngineID UInt8,  IsArtifical UInt8,  WindowClientWidth UInt16,  WindowClientHeight UInt16,  ClientTimeZone Int16,  ClientEventTime DateTime,  SilverlightVersion1 UInt8, SilverlightVersion2 UInt8,  SilverlightVersion3 UInt32,  SilverlightVersion4 UInt16,  PageCharset String,  CodeVersion UInt32,  IsLink UInt8,  IsDownload UInt8,  IsNotBounce UInt8,  FUniqID UInt64,  HID UInt32,  IsOldCounter UInt8, IsEvent UInt8,  IsParameter UInt8,  DontCountHits UInt8,  WithHash UInt8, HitColor FixedString(1),  UTCEventTime DateTime,  Age UInt8,  Sex UInt8,  Income UInt8,  Interests UInt16,  Robotness UInt8,  GeneralInterests Array(UInt16), RemoteIP UInt32,  RemoteIP6 FixedString(16),  WindowName Int32,  OpenerName Int32,  HistoryLength Int16,  BrowserLanguage FixedString(2),  BrowserCountry FixedString(2),  SocialNetwork String,  SocialAction String,  HTTPError UInt16, SendTiming Int32,  DNSTiming Int32,  ConnectTiming Int32,  ResponseStartTiming Int32,  ResponseEndTiming Int32,  FetchTiming Int32,  RedirectTiming Int32, DOMInteractiveTiming Int32,  DOMContentLoadedTiming Int32,  DOMCompleteTiming Int32,  LoadEventStartTiming Int32,  LoadEventEndTiming Int32, NSToDOMContentLoadedTiming Int32,  FirstPaintTiming Int32,  RedirectCount Int8, SocialSourceNetworkID UInt8,  SocialSourcePage String,  ParamPrice Int64, ParamOrderID String,  ParamCurrency FixedString(3),  ParamCurrencyID UInt16, GoalsReached Array(UInt32),  OpenstatServiceName String,  OpenstatCampaignID String,  OpenstatAdID String,  OpenstatSourceID String,  UTMSource String, UTMMedium String,  UTMCampaign String,  UTMContent String,  UTMTerm String, FromTag String,  HasGCLID UInt8,  RefererHash UInt64,  URLHash UInt64,  CLID UInt32,  YCLID UInt64,  ShareService String,  ShareURL String,  ShareTitle String,  ParsedParams Nested(Key1 String,  Key2 String, Key3 String, Key4 String, Key5 String,  ValueDouble Float64),  IslandID FixedString(16),  RequestNum UInt32,  RequestTry UInt8')
WHERE URL != '';


SELECT URL, count(URL) as Count
FROM hits_NoPrimaryKey
WHERE UserID = 749927693
GROUP BY URL
ORDER BY Count DESC
LIMIT 10;


┌─URL──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┬─Count─┐
│ http://auto.ru/chatay-barana.ru/traction.html#maybettaya                                                                                                                                                         │   170 │
│ http://auto.ru/chatay-id=371&rents/41601151                                                                                                                                                                      │    52 │
│ http://public_search                                                                                                                                                                                             │    45 │
│ http://kovrik-medvedevushku-zarabotaet-davi-youer-glava-15-tsentmenyuk-ozerom-bolshoyvoprosofia.ru/pesni-politics/20/03/2013/2014/9119915412/commercedes-benchange_kuhnyaya%2F&is_vative=0&search?cl4url=ya-mail │    36 │
│ http://forumal                                                                                                                                                                                                   │    33 │
│ http://korablitz.ru/L_1OFFERS_CRD                                                                                                                                                                                │    14 │
│ http://auto.ru/chatay-id=371&rents/4160115                                                                                                                                                                       │    14 │
│ http://auto.ru/chatay-john-Den-Yunan ahtım var b                                                                                                                                                                 │    13 │
│ http://auto.ru/chatay-john-Den-Yunan ahtım var&is                                                                                                                                                                │    10 │
│ http://wot/html?page/23600_m/sportbox.ru/audiovkontur.ru/articles                                                                                                                                                │     9 │
└──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┴───────┘

10 rows in set. Elapsed: 0.068 sec. Processed 8.87 million rows, 68.85 MB (129.77 million rows/s., 1.01 GB/s.)
```

**Comment by DanRoscigno** [comment]
@tom-clickhouse 
I reproduced it 2 weeks ago, and again just now. I noticed that you used a different query to load the data, is that equivalent to this one from the docs:
```
INSERT INTO hits_NoPrimaryKey SELECT
   intHash32(c11::UInt64) AS UserID,
   c15 AS URL,
   c5 AS EventTime
FROM url('https://datasets.clickhouse.com/hits/tsv/hits_v1.tsv.xz')
WHERE URL != '';
```

**Comment by tom-clickhouse** [comment]
For some reason `FROM url('https://datasets.clickhouse.com/hits/tsv/hits_v1.tsv.xz')` completely fails in the newest version.
Whereas if you provide full schema like in my example, it works.

Already started a branch updating the data load query.

**Comment by tom-clickhouse** [comment]
Fixed by #642
